ansible/roles/software/ckan/inventory/tasks/datapusher-next.yml

```diff
@@ -12,54 +12,28 @@
       - libxslt1-dev
       - libxml2-dev
       - git
+      - libffi-dev
 
 - name: Remove old datapusher code
   file:
     path: "{{ datapusher_virtual_env }}"
     state: absent
 
-- name: setup virtualenv for new datapusher
-  pip:
-    name:
-      - pip>=19.3.0
-      - setuptools>=44.0.0,<45.0.0  # setuptools 45+ requires python 3.5, keep pinned below, but still modern version
-    virtualenv: '{{ datapusher_virtual_env }}'
-    umask: "0022"
-  notify: restart apache2
-
-- name: setup gunicorn for new datapusher
-  pip:
-    name:
-      - gunicorn~=19.10.0
-      - gevent~=20.6.2
-    virtualenv: '{{ datapusher_virtual_env }}'
-    umask: "0022"
-
-- name: Make datapusher source code directory
-  file:
-    path: "{{ datapusher_virtual_env }}/src"
-    state: directory
-
-- name: Git clone datapusher
-  git:
-    force: yes
-    umask: "0022"
-    repo: "{{ datapusher_next_repo }}"
-    dest: "{{ datapusher_virtual_env }}/src"
-    version: "{{ datapusher_next_branch }}"
+- include_tasks: build-pkgs.yml
+  vars:
+    build_pkg_name: "{{ datapusher_build_pkg_name }}"
+    build_pkg_branch: "{{ datapusher_build_pkg_branch }}"
+    build_pkg_repo: "{{ datapusher_build_pkg_repo }}"
+    build_pkg_requirements: "{{ datapusher_next_requirements }}"
+    virtualenv: "{{ datapusher_virtual_env }}"
 
-- name: install requirements for datapusher
+- name: install psycopg2
   pip:
-    requirements: "{{ datapusher_virtual_env }}/src/{{ datapusher_next_requirements }}"
+    name: "psycopg2"
```

Comment:
Why not install this from a requirements.txt?

Reply:
This was built to use upstream, which does not contain psycopg2 but is required for how we setup our virtual environment. We will need to investigate why it is necessary for our virtual environment, and clean up or submit a PR upstream. Either way, leaving this breadcrumb to note that we shouldn't support this long term.